include/CLI/App.hpp

```diff
@@ -2116,52 +2234,59 @@ class App {
                 opt->add_result(positional);
                 parse_order_.push_back(opt.get());
                 args.pop_back();
-                return;
+                return true;
             }
         }
 
         for(auto &subc : subcommands_) {
             if((subc->name_.empty()) && (!subc->disabled_)) {
-                subc->_parse_positional(args);
-                if(subc->missing_.empty()) { // check if it was used and is not in the missing category
-                    return;
-                } else {
-                    args.push_back(std::move(subc->missing_.front().second));
-                    subc->missing_.clear();
+                if(subc->_parse_positional(args)) {
+                    if(!subc->pre_parse_called_) {
+                        subc->_trigger_pre_parse(args.size());
+                    }
+                    return true;
                 }
             }
         }
-
+        /// let the parent deal with it if possible
```

Comment:
Stylistic note: Comments inline are two `//`, three slashes indicated that Doxygen picks up the comment for the API docs, and it can't pick up comments in code.